### PR TITLE
Define what's in scope for OpenTelemetry Logs GA (post traces and metrics 1.0 GA)

### PR DESCRIPTION
**Pull request by tigrannajaryan**
Clearly defined scope is important to align all logs contributors and to make
sure we know what target we are working towards. Note that some of the listed
items are already fully or partially implemented but are still listed for
completeness.

This OTEP shows the initial scope proposal. Please suggest additions or changes
to the scope and I will include them in this PR.

### CODE REVIEW

File: text/logs/0130-logs-1.0ga-definition.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+### Specification and SDK
+
+- Write guidelines and specification for logging libraries to support
+  OpenTelemetry-compliant logs.
+
+- Implement OpenTelemetry logs SDK for Java that support trace context
+  extraction from the current execution context.
+
+- Implement appenders for Log4J using OpenTelemetry SDK that automatically
+  includes extracted trace context in the logs and supports exporting via OTLP.
```
Comment (yurishkuro) [inline text/logs/0130-logs-1.0ga-definition.md:23]: Why appenders?  With OpenTracing we opted for implementing a scope manager that ensures that slf4j.MDC is updated on every span context switch. Then it becomes trivial to include context fields into the logs.

Comment (tigrannajaryan) [reply text/logs/0130-logs-1.0ga-definition.md:23]: @zenmoto can you comment on this? You did the proof of concept and you know more about this than me.

File: text/logs/0130-logs-1.0ga-definition.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+### Specification and SDK
+
+- Write guidelines and specification for logging libraries to support
+  OpenTelemetry-compliant logs.
+
+- Implement OpenTelemetry logs SDK for Java that support trace context
+  extraction from the current execution context.
+
+- Implement appenders for Log4J using OpenTelemetry SDK that automatically
```
Comment (iNikem) [inline text/logs/0130-logs-1.0ga-definition.md:22]: Why Log4J and not SLF4J

File: text/logs/0130-logs-1.0ga-definition.md (hunk 1 above)
Comment (danielwegener) [reply text/logs/0130-logs-1.0ga-definition.md:23]: As far as I understood, the only MDC scope manager wrapper I've found has been implemented in jaeger-client-java directly https://github.com/jaegertracing/jaeger-client-java/pull/718.

File: text/logs/0130-logs-1.0ga-definition.md (hunk 2 above)
Comment (danielwegener) [reply text/logs/0130-logs-1.0ga-definition.md:22]: appenders are not part of the slf4j api, they are implementor specific (e.g. ch.qos.logback.core.Appender or org.apache.logging.log4j.core.Appender)

Comment (tigrannajaryan) [reply text/logs/0130-logs-1.0ga-definition.md:22]: I have no particular preference for which Java logging library to support. I can change to SFL4J if it is a more popular/preferable library for the community.
The goal here is not to have comprehensive coverage of all Java application needs, but rather to set an example and clearly demonstrate how OpenTelemetry logging SDK can be used with a logging library.

Comment (bogdandrutu) [reply text/logs/0130-logs-1.0ga-definition.md:22]: I would say appenders for at least one of the major logging library?

File: text/logs/0130-logs-1.0ga-definition.md [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+## Logs 1.0 GA Todo List
+
+### Specification and SDK
+
+- Write guidelines and specification for logging libraries to support
+  OpenTelemetry-compliant logs.
+
+- Implement OpenTelemetry logs SDK for Java that support trace context
+  extraction from the current execution context.
```
Comment (jkwatson) [inline text/logs/0130-logs-1.0ga-definition.md:20]: Was this discussed with the maintainers of the Java project?

Comment (tigrannajaryan) [reply text/logs/0130-logs-1.0ga-definition.md:20]: > I had the understanding the Logs portion of OpenTelemetry was explicitly not a part of the upcoming GA. Has that changed?

@MrAlias GA here means "whenever we decide to GA logs support", not the upcoming GA where only trace and metrics will be supported. I should probably remove the reference to 1.0 to avoid confusion.

**Comment by tigrannajaryan** [comment]
> I had the understanding the Logs portion of OpenTelemetry was explicitly not a part of the upcoming GA. Has that changed?

@MrAlias GA here means "whenever we decide to GA logs support", not the upcoming GA where only trace and metrics will be supported. I should probably remove the reference to 1.0 to avoid confusion.

File: text/logs/0130-logs-1.0ga-definition.md (hunk 3 above)
Comment (tigrannajaryan) [reply text/logs/0130-logs-1.0ga-definition.md:20]: Not yet, this is the opening of the discussion :-)
We have a logs SIG meeting tomorrow, Java maintainers can join to discuss, or log people can join the next Java SIG, whatever works best.

File: text/logs/0130-logs-1.0ga-definition.md (hunk 2 above)
Comment (tigrannajaryan) [reply text/logs/0130-logs-1.0ga-definition.md:22]: That's a good  point, let me rephrase to avoid mentioning a specific library. The decision on which library to support and how exactly does not need to happen right now.

**Comment by MrAlias** [comment]
> > I had the understanding the Logs portion of OpenTelemetry was explicitly not a part of the upcoming GA. Has that changed?
> 
> @MrAlias GA here means "whenever we decide to GA logs support", not the upcoming GA where only trace and metrics will be supported. I should probably remove the reference to 1.0 to avoid confusion.

Ah, gotcha. Thanks for the clarification :+1:

File: text/logs/0130-logs-1.0ga-definition.md (hunk 3 above)
Comment (jkwatson) [reply text/logs/0130-logs-1.0ga-definition.md:20]: I'm completely swamped right now, so if logs people could join the Java SIG, that would be best. My main concern is distracting/eating time from from the rest of the GA effort, and who will maintain the submitted log-oriented code in the meantime.

Comment (tigrannajaryan) [reply text/logs/0130-logs-1.0ga-definition.md:20]: I definitely don't want to distract Java SIG. We do not need to make detailed decisions now. The purpose of this PR is primarily to align people who already already are working on logs and create clarity around the goals. We will not be asking current Java SIG members to do any actual work for logs until traces and metrics are fully done and ready for GA.

However, if we can agree on the goals for logs I hope you guys don't mind if the alpha implementation of the logs is done in parallel by other contributors and we don't stall it for several months.

I'll join one of the coming Java SIG meetings to discuss this. @zenmoto would be great if you can join too. Let's sync.

File: text/logs/0130-logs-1.0ga-definition.md (hunk 1 above)
Comment (yurishkuro) [reply text/logs/0130-logs-1.0ga-definition.md:23]: @danielbang907 that's a very recent addition, there is way plenty of prior art. 

In any case, I agree that it's best to simply remove the reference to a concrete logging library, as well as the reference to "appenders", because that's just one way to achieve that, and just say "integration with major logging libraries". Implementation details are not relevant to the goals.

File: text/logs/0130-logs-1.0ga-definition.md (hunk 2 above)
Comment (tigrannajaryan) [reply text/logs/0130-logs-1.0ga-definition.md]: Re-worded, please check again.

**Comment by tigrannajaryan** [comment]
@MrAlias I clarified what "GA" means in the doc.

File: text/logs/0130-logs-1.0ga-definition.md (hunk 1 above)
Comment (tigrannajaryan) [reply text/logs/0130-logs-1.0ga-definition.md]: Removed the reference to "appenders", please check.

File: text/logs/0130-logs-1.0ga-definition.md (hunk 3 above)
Comment (jkwatson) [reply text/logs/0130-logs-1.0ga-definition.md:24]: We've already been getting PRs from @zenmoto in the Java repo that is distracting me from other work, so yes, it would be great if y'all could come to the Java SIG and talk about the plan. Thanks!

File: text/logs/0130-logs-1.0ga-definition.md (hunk 1 above)
Comment (zenmoto) [reply text/logs/0130-logs-1.0ga-definition.md]: Sorry for the delay in getting back to this. It really depends on the logging library being adapted- Logback and Log4J have an MDC concept, I don't believe that the Java logging library does, and we need to be able to work in non-Java environments, too. I agree with removing a reference to appenders, perhaps something generic like "extension mechanisms" would be broader and more inclusive of multiple approaches.

File: text/logs/0130-logs-1.0ga-definition.md (hunk 2 above)
Comment (zenmoto) [reply text/logs/0130-logs-1.0ga-definition.md]: I think that it's important to have a solution that works generally- in the Java world, at least Log4j, Log4j2, Logback, and java.util.logging (though obviously we start with one).

File: text/logs/0130-logs-1.0ga-definition.md (hunk 3 above)
Comment (zenmoto) [reply text/logs/0130-logs-1.0ga-definition.md:24]: Unfortunately I will be unable to make this week's Java SIG, but I will plan to join next week (and I'll talk with @tigrannajaryan afterwards).

Comment (tigrannajaryan) [reply text/logs/0130-logs-1.0ga-definition.md:24]: Next week is fine.

File: text/logs/0130-logs-1.0ga-definition.md (hunk 2 above)
Comment (tigrannajaryan) [reply text/logs/0130-logs-1.0ga-definition.md]: Sounds good, I think the text now reflects this.

File: text/logs/0130-logs-1.0ga-definition.md (hunk 1 above)
Comment (tigrannajaryan) [reply text/logs/0130-logs-1.0ga-definition.md]: Removed. Resolving this comment, please reopen if still not clear enough.

File: text/logs/0130-logs-1.0ga-definition.md [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+  of OTLP but is not supported by the Collector yet).
+
+- Implement log exporters in Collector for a few vendor formats from
+  participating vendors.
+
+- Implement Fluent Forward protocol receiver to receive logs from
+  FluentBit/FluentD.
+
+- Add support for log data type to the following processors: `resource`,
+  `batch`, `attributes`, `k8s_tagger`, `resourcedetection`.
```
Comment (pmm-sumo) [inline text/logs/0130-logs-1.0ga-definition.md:51]: What about `queued` (Queued Retry)?

Comment (davidblondeau) [reply text/logs/0130-logs-1.0ga-definition.md:51]: same question for `filter`?

Comment (tigrannajaryan) [reply text/logs/0130-logs-1.0ga-definition.md:51]: queued_retry is [being deprecated](https://github.com/open-telemetry/opentelemetry-collector/pull/1386) and the equivalent functionality is moved to exporters (and support for logs already exists).

Comment (tigrannajaryan) [reply text/logs/0130-logs-1.0ga-definition.md:51]: `filter` is currently a metric-only processor. Not sure if logs support needs to be part of the GA. We probably need to add it for logs, but first it will probably happen for traces.
